source/common/router/config_utility.h

```diff
@@ -44,6 +45,21 @@ class ConfigUtility {
     const bool is_regex_;
   };
 
+  // A QueryParameterMatcher specifies one "name" or "name=value" element
+  // to match in a request's query string. It is the optimized, runtime
+  // equivalent of the QueryParameterMatcher proto in the RDS v2 API.
+  struct QueryParameterMatcher {
+    QueryParameterMatcher(const envoy::api::v2::QueryParameterMatcher& config)
+        : name_(config.name()), value_(config.value()),
+          regex_pattern_(value_, std::regex::optimize),
+          is_regex_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, regex, false)) {}
+
+    const std::string name_;
```

Comment:
Can some of the inner logic in `ConfigUtility::matchQueryParams` be moved to a method here?

Reply:
Sounds good; I've just posted an update that encapsulates the logic in this object. We should probably do the same with `struct HeaderData` and `ConfigUtility::matchHeaders`, but I'll make a separate PR for that.